application/org.openjdk.jmc.rjmx/src/main/java/org/openjdk/jmc/rjmx/ConnectionToolkit.java

```diff
@@ -365,6 +365,20 @@ public static boolean isHotSpot(IConnectionHandle connectionHandle) {
 		return vmName != null && JavaVMVersionToolkit.isHotspotJVMName(vmName);
 	}
 
+	/**
+	 * Returns {@code true} if the connection handle is connected to a Substrate VM, {@code false}
+	 * otherwise. This method requires the connection handle to be connected.
+	 *
+	 * @param connectionHandle
+	 *            the connection handle to check.
+	 * @return {@code true} if the connection handle is connected to a Substrate VM, {@code false}
+	 *         otherwise.
+	 */
+	public static boolean isSubstrateVM(IConnectionHandle connectionHandle) {
+		String vmName = getVMName(connectionHandle);
```

Comment:
There's also a null check in JavaVMVersionToolkit.isSubstrateVMName, so we can skip it here. But probably it's ok to keep as well.

Reply:
Nice catch, although it's setup this way for similar functions so I might side with keeping it as is. The difference being that with ConnectionToolkit we can check the VM given a connection handle, whereas JavaVMVersionToolkit just does the string comparison. It also just so happens that ConnectionToolkit will grab the vm name and delegate to JavaVMVersionToolkit, but they're both public and suit different needs.

`JVMType.getJVMType()` uses the `JavaVMVersionToolkit.is*VMName()` and it might be nice to keep the null check for any potential problems there.